### PR TITLE
ABLASTR: Move TextMsg

### PR DESCRIPTION
**Pull request by ax3l**
Since this is used in a ABLASTR header, we need to compile this in ABLASTR, so ImpactX can use it as well. Otherwise we have missing symbols for that utility.

Follow-up to #2939 (& #2952)

```
/home/runner/work/impactx/impactx/build/_deps/fetchedablastr-src/Source/ablastr/particles/DepositCharge.H:83: undefined reference to `Utils::TextMsg::Assert(char const*, char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: /home/runner/work/impactx/impactx/build/_deps/fetchedablastr-src/Source/ablastr/particles/DepositCharge.H:92: undefined reference to `Utils::TextMsg::Assert(char const*, char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: /home/runner/work/impactx/impactx/build/_deps/fetchedablastr-src/Source/ablastr/particles/DepositCharge.H:98: undefined reference to `Utils::TextMsg::Assert(char const*, char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
/usr/bin/ld: /home/runner/work/impactx/impactx/build/_deps/fetchedablastr-src/Source/ablastr/particles/DepositCharge.H:102: undefined reference to `Utils::TextMsg::Assert(char const*, char const*, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)'
collect2: error: ld returned 1 exit status
```

### CODE REVIEW

**Review by lucafedeli88 — APPROVED** [review]
Thanks Axel!

File: Source/ablastr/utils/Make.package [hunk 1]
```diff
@@ -0,0 +1,3 @@
+CEXE_sources += TestMsg.cpp
```
Comment (lucafedeli88) [inline Source/ablastr/utils/Make.package]:
```suggestion
CEXE_sources += TextMsg.cpp
```

**Review by lucafedeli88 — DISMISSED** [review]
There is a small typo, otherwise looks fine to me!

File: Source/ablastr/utils/Make.package (hunk 1 above)
Comment (ax3l) [reply Source/ablastr/utils/Make.package]: Thanks, that explains why CI fails :D

File: Source/ablastr/utils/TextMsg.H [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        const std::string& text,
+        const int max_line_length);
+
+namespace TextMsg
+{
+    /** \brief This function formats a text message as an error message,
+    *   adding the '### ERROR: ' prefix and (by default) performing text wrapping
+    *
+    * @param[in] msg the string to be formatted
+    * @param[in] do_text_wrapping if true, the message is automatically wrapped
```
Comment (ax3l) [inline Source/ablastr/utils/TextMsg.H]:
```suggestion
    * @param[in] do_text_wrapping if true, the text of the message is automatically wrapped
```

File: Source/ablastr/utils/TextMsg.H [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    * @return the formatted message
+    */
+    std::string
+    Err (const std::string &msg, const bool do_text_wrapping = true);
+
+    /** \brief This function formats a text message as an info message,
+    *   adding the '### INFO: ' prefix and (by default) performing text wrapping
+    *
+    * @param[in] msg the string to be formatted
+    * @param[in] do_text_wrapping if true, the message is automatically wrapped
```
Comment (ax3l) [inline Source/ablastr/utils/TextMsg.H]:
```suggestion
    * @param[in] do_text_wrapping if true, the text of the message is automatically wrapped
```

File: Source/ablastr/utils/TextMsg.H [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    std::string
+    Info (const std::string &msg, const bool do_text_wrapping = true);
+
+    /** \brief This function formats a text message as a warning message,
+    *   adding the '### WARN: ' prefix and (by default) performing text wrapping.
+    *   Warning: this format is not used by the WarningLogger, which has an internal,
+    *   dedicated, formatter.
+    *
+    * @param[in] msg the string to be formatted
+    * @param[in] do_text_wrapping if true, the message is automatically wrapped
```
Comment (ax3l) [inline Source/ablastr/utils/TextMsg.H]:
```suggestion
    * @param[in] do_text_wrapping if true, the text of the message is automatically wrapped
```

**Review by ax3l — COMMENTED** [review]
Apply exact wording from  #2946

**Comment by ax3l** [comment]
@lucafedeli88 ready for re-approval :)

**Review by lucafedeli88 — APPROVED** [review]
Thanks for this PR @ax3l !

**Pull request merged**